test/functional/mining_mainnet.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test mining on an alternate mainnet
+
+Test mining related RPCs that involve difficulty adjustment, which
+regtest doesn't have.
+
+It uses an alternate mainnet chain. See data/README.md for how it was generated.
+
+Mine one retarget period worth of blocks with a short interval in
+order to maximally raise the difficulty. Verify this using the getmininginfo RPC.
+
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+from test_framework.blocktools import (
+    DIFF_1_N_BITS,
+    DIFF_1_TARGET,
+    DIFF_4_N_BITS,
+    DIFF_4_TARGET,
+    create_coinbase,
+    nbits_str,
+    target_str
+)
+
+from test_framework.messages import (
+    CBlock,
+)
+
+import json
+import os
+
+# Derived from first BIP32 test vector master key:
+# Use pkh() because tr() outputs at low heights are not spendable (unexpected-witness)
+COINBASE_OUTPUT_DESCRIPTOR="pkh(xprv9s21ZrQH143K3QTDL4LXw2F7HEK3wJUD2nW2nRk4stbPy6cq3jPPqjiChkVvvNKmPGJxWUtg6LnF5kejMRNNU3TGtRBeJgk33yuGBxrMPHi/44h/0h/0h/<0;1>/*)#fkjtr0yn"
+COINBASE_SCRIPT_PUBKEY="76a914eadbac7f36c37e39361168b7aaee3cb24a25312d88ac"
```

Comment:
In commit "test: check difficulty adjustment using alternate mainnet" (90b6c189d9b18203e119a5f3478f7b0da7d71392)

Note: this seems ok, but I guess it could be simplified a little since COINBASE_OUTPUT_DESCRIPTOR is unused, and no spending is done anymore.

In case you do want to bring back the spending test that seems fine if it's explained a little. I wasn't necessarily asking to remove it before, but was just confused by why it was there and a little overwhelmed with the details chain generation in the test description. Keeping this is as or simplifying more also both seem fine to me.

Reply:
I'm fine with leaving it out.

I also dropped the `COINBASE_OUTPUT_DESCRIPTOR` constant and moved the descriptor itself to the README.